hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/async/RouterAsyncClientProtocol.java

```diff
@@ -355,7 +355,6 @@ public boolean mkdirs(String src, FsPermission masked, boolean createParent)
       return rpcClient.invokeAll(locations, method);
     }
 
-    asyncComplete(false);
```

Comment:
Hi, @hfutatzhanghb  I just took a look at this method, and I think it should work fine without the modifications you suggested. First, set the completeFuture in the thread pool variable to false. Next, if locations.size() > 1, the completeFuture after execution will replace the thread variable. If locations.size() <= 1, then since the initial value of completeFuture is set to false, the subsequent asynchronous call code will be executed.

Reply:
Hi, @KeeProMise , thanks for your reviewing.  Have reverted mkdirs and other rpcs not affected by asyncComplete method to reduce code duplication. Thanks again.